src/libraries/System.Data.OleDb/src/OleDbDataAdapter.COMWrappers.cs

```diff
@@ -0,0 +1,154 @@
+                    string? tmp = null;
+                    if (data is DataSet)
+                    {
+                        tmp = GetSourceTableName(srcTable!, resultCount);
+                    }
+                    results += FillFromRecordset(data, recordset, tmp, out incrementResultCount);
+
+                    if (multipleResults)
+                    {
+                        value[0] = DBNull.Value;
+
+                        object recordsAffected;
+                        object nextresult;
+                        OleDbHResult hr = ((UnsafeNativeMethods.Recordset15)adodb).NextRecordset(out recordsAffected, out nextresult);
+
+                        if (0 > hr)
+                        {
+                            // Current provider does not support returning multiple recordsets from a single execution.
+                            if (ODB.ADODB_NextResultError != (int)hr)
+                            {
+                                IntPtr pErrorInfo;
+                                UnsafeNativeMethods.GetErrorInfo(0, out pErrorInfo);
+
+                                string message = string.Empty;
+                                throw new COMException(message, (int)hr);
+                            }
+                            break;
+                        }
+                        adodb = nextresult;
+                        if (null != adodb)
+                        {
+                            recordset = (UnsafeNativeMethods.ADORecordsetConstruction)adodb;
+
+                            if (incrementResultCount)
+                            {
+                                resultCount++;
+                            }
+                            continue;
+                        }
+                    }
+                    break;
+                } while (null != recordset);
+
+                if ((null != recordset) && (closeRecordset || (null == adodb)))
+                {
+                    FillClose(true, recordset);
+                }
+            }
+            else if (null != record)
+            {
+                results = FillFromRecord(data, record, srcTable!);
+                if (closeRecordset)
+                {
+                    FillClose(false, record);
+                }
+            }
+            else
+            {
+                throw ODB.Fill_NotADODB("adodb");
+            }
+            return results;
+        }
+
+        private void FillClose(bool isrecordset, object value)
+        {
+            OleDbHResult hr;
+            if (isrecordset)
+            {
+                hr = ((UnsafeNativeMethods.Recordset15)value).Close();
+            }
+            else
+            {
+                hr = ((UnsafeNativeMethods._ADORecord)value).Close();
+            }
+            if ((0 < (int)hr) && (ODB.ADODB_AlreadyClosedError != (int)hr))
+            {
+                IntPtr pErrorInfo;
+                UnsafeNativeMethods.GetErrorInfo(0, out pErrorInfo);
```

Comment:
Do we not do anything with the `IErrorInfo`? Is this simply used to clear it?

Reply:
Same above. If this is only for clearing we have an API to skip the query of `IErrorInfo`.